conan/tools/env/environment.py

```diff
@@ -351,3 +344,21 @@ def loads(text):
             else:
                 raise ConanException("Bad env defintion: {}".format(line))
         return result
+
+
+def save_script(conanfile, env, name):
+    # FIXME: using platform is not ideal but settings might be incomplete
+    if platform.system() == "Windows":
+        path = os.path.join(conanfile.generators_folder, "{}.bat".format(name))
+        env.save_bat(path)
+    else:
+        path = os.path.join(conanfile.generators_folder, "{}.sh".format(name))
+        env.save_sh(path)
+
+    register_environment_script(conanfile, path)
+
+
+def register_environment_script(conanfile, path):
+    element = os.path.basename(path)
```

Comment:
Are we registering only the names, not the paths to the scripts? Then assume it will always be relative to generators_folder? What if some tool reference some script that is in the profiles folder?

Reply:
I didn't think about this being a public function and not sure about accepting scripts in the profile folder, but ok, we can add the full path. Just a bit more ugly if someone wants to "alter" the list by hand.